internal/docker/deployment.go

```diff
@@ -36,8 +36,10 @@ type HomeserverDeployment struct {
 	accessTokensMutex   sync.RWMutex
 	ApplicationServices map[string]string // e.g { "my-as-id": "id: xxx\nas_token: xxx ..."} }
 	DeviceIDs           map[string]string // e.g { "@alice:hs1": "myDeviceID" }
-	CSAPIClients        []*client.CSAPI
-	CSAPIClientsMutex   sync.Mutex
+
+	// track all clients so if Restart() is called we can repoint to the new high-numbered port
```

Comment:
I'm failing to understand this comment, but maybe I'll get it once I'm further in the review